File: dbflow/src/main/java/com/raizlabs/android/dbflow/structure/cache/SimpleMapCache.java
package com.raizlabs.android.dbflow.structure.cache;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.raizlabs.android.dbflow.config.FlowLog;
import com.raizlabs.android.dbflow.structure.Model;

import java.util.HashMap;
import java.util.Map;

/**
 * Description: A simple implementation that keeps {@link Model} you interact with in memory.
 */
public class SimpleMapCache<TModel> extends ModelCache<TModel, Map<Object, TModel>> {

    /**
     * Constructs new instance with a {@link HashMap} with the specified capacity.
     *
     * @param capacity The capacity to use on the hashmap.
     */
    public SimpleMapCache(int capacity) {
        super(new HashMap<Object, TModel>(capacity));
    }

    /**
     * Constructs new instance with a cache
     *
     * @param cache The arbitrary underlying cache class.
     */
    public SimpleMapCache(@NonNull Map<Object, TModel> cache) {
        super(cache);
    }

    @Override
    public void addModel(@Nullable Object id, @NonNull TModel model) {
        getCache().put(id, model);
    }

    @Override
    public TModel removeModel(@NonNull Object id) {
        return getCache().remove(id);
    }

    @Override
    public void clear() {
        getCache().clear();
    }

    @Override
    public TModel get(@Nullable Object id) {
        return getCache().get(id);
    }

    @Override
    public void setCacheSize(int size) {
        FlowLog.log(FlowLog.Level.I, "The cache size for " + SimpleMapCache.class.getSimpleName() +
                " is not re-configurable.");
    }
}

File: dbflow-common/src/main/kotlin/com/raizlabs/dbflow5/query/cache/SimpleMapCache.kt
package com.raizlabs.dbflow5.query.cache

import com.raizlabs.dbflow5.config.FlowLog
import com.raizlabs.dbflow5.structure.Model

/**
 * Description: A simple implementation that keeps [Model] you interact with in memory.
 */
class SimpleMapCache<TModel> : ModelCache<TModel, MutableMap<Any?, TModel>> {

    /**
     * Constructs new instance with a [HashMap] with the specified capacity.
     *
     * @param capacity The capacity to use on the hashmap.
     */
    constructor(capacity: Int) : super(HashMap<Any?, TModel>(capacity))

    /**
     * Constructs new instance with a cache
     *
     * @param cache The arbitrary underlying cache class.
     */
    constructor(cache: MutableMap<Any?, TModel>) : super(cache)

    override fun addModel(id: Any?, model: TModel) {
        cache.put(id, model)
    }

    override fun removeModel(id: Any): TModel? = cache.remove(id)

    override fun clear() {
        cache.clear()
    }

    override fun get(id: Any?): TModel? = cache[id]

    override fun setCacheSize(size: Int) {
        FlowLog.log(FlowLog.Level.W, "The cache size for SimpleMapCache is not re-configurable.")
    }
}
